torch/_decomp/__init__.py

```diff
@@ -263,170 +263,51 @@ def remove_decompositions(
-    for op in cia_ops:
-        split_list = op.split(".")
-        # Sometime overload could be missing
-        assert len(split_list) == 1 or len(split_list) == 2
-        op_name = split_list[0]
-        op_overload_name = "default"
-        if len(split_list) == 2:
-            op_overload_name = split_list[1]
-
-        _ = getattr(getattr(torch.ops.aten, op_name), op_overload_name)
-
-    # Second step to finally compile the list of all valid ops
-    cia_ops = set()
-    for op in torch.ops.aten:
-        op_packet = getattr(torch.ops.aten, op)
-        for overload in op_packet.overloads():
-            op_overload = getattr(op_packet, overload)
-            if _is_preservable_cia_op(op_overload):
-                cia_ops.add(op_overload)
-    return cia_ops
-
-
-def _get_decomp_for_cia(op):
-    # [NOTE] Seperating out func.decompose
-    # Ideally we should be able to just register func.decompose but
-    # we can't as this decomp is gonna be registered to the py_impl.
-    # As a result it will infinitely recurse. So we first check if the op
-    # has py_impl entry for CIA and if it is we use that first. If not,
-    # we register C++ query to py_impl.
-    dk = torch._C.DispatchKey.CompositeImplicitAutograd
-    if dk in op.py_kernels and not isinstance(op.py_kernels[dk], torch._C.DispatchKey):
-        return op.py_kernels[dk]
-
-    def _special_op_to_decompose_cia(*args, **kwargs):
-        kernel = kwargs["kernel"]
-        del kwargs["kernel"]
-        # Can't call kernel.decompose due to infinite recursion as
-        # we register this kernel to py_impl directly
-        dk = torch._C.DispatchKey.CompositeImplicitAutograd
-        if torch._C._dispatch_has_kernel_for_dispatch_key(
-            kernel.name(), torch._C.DispatchKey.CompositeImplicitAutograd
-        ):
-            return kernel._op_dk(dk, *args, **kwargs)
-        else:
-            raise AssertionError(
-                f"Expected {kernel} to have CompositeImplicitAutograd kernel"
-            )
-
-    return partial(_special_op_to_decompose_cia, kernel=op)
-
-
 # See NOTE [Core ATen Ops]
 #
 # list was copied from torch/_inductor/decomposition.py
 # excluding decompositions that results in prim ops
 # Resulting opset of decomposition is core aten ops
 def core_aten_decompositions() -> Dict[torch._ops.OperatorBase, Callable]:
-    decomp_table = _core_aten_decompositions_post_autograd()
-
     # If it is fbcode change, we return the old decomposition list
+    from torch._export.utils import (
+        _collect_all_valid_cia_ops_for_aten_namespace,
+        _get_decomp_for_cia,
+    )
     from torch._inductor import config
 
+    aten = torch.ops.aten
+
+    # Entry without functional CIA ops
+    decomp_table = _core_aten_decompositions_post_autograd()
     if config.is_fbcode():
-        return decomp_table
+        decomp_table_for_functional_cia_ops_legacy = get_decompositions(
+            [
+                aten.all.dimname,
```

Comment:
Can you help me understand the migration story here: it looks like prior to this PR, this extra list of decomps used to be in the `not config.is_fbcode()` path, but now it is flipped (it only shows up in the `is_fbcode()` path)? Or am I mis-reading

(also, it looks like you dropped the comment associated with the list on like 399)

Reply:
So what's happening here is:
1. The clean list without any functional CIA is in _core_aten_decompositions_post_autograd(). 
2. Previously, if it is fbcode, we would try to return the decomp table without any functional CIA ops due to my changes like 3 weeks ago. But in this PR, I am trying to go WAY BACK to make sure the core_aten_decomp has the entries for those ignored functional CIA as well. I think it makes no difference tho in practice

I will add the comment back haha

Reply:
(agreed offline to remove the extra decomps)